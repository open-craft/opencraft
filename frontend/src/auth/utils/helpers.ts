import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
