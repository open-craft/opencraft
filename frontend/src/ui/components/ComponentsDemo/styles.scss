@import '~styles/theme';

.components-demo-page {
  width: 100%;

  .components-demo-heading {
    padding: 1rem;
  }

  h1,
  h2 {
    text-align: center;
    color: #fff;
  }

  h1 {
    font-family: $playfair-font;
    font-weight: bold;
    font-size: 40px;
  }

  h2 {
    font-family: $chivo-font;
    font-size: 20px;
    margin-bottom: 4rem;
  }

  .components-demo-container {
    background-color: #fff;
    padding: 4rem 1rem;
    margin: 0 -15px;
    flex: 1;

    .components-demo-text {
      margin-bottom: 1rem;
      margin-left: 1rem;
    }

    .collapse-edit-area {
      padding: 0rem 2rem;
    }

    .color-input-field {
      padding: 2rem;
    }

    .table-cell {
      padding: 1rem;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }

  .components-demo-table-container {
    margin: 0 1rem;
  }
}
