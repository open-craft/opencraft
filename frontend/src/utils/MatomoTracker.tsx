import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
