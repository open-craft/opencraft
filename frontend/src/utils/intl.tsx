import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
