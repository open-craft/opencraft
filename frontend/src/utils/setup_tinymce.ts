import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
