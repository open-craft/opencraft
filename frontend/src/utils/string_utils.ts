export const toCamelCase = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace('-', '').replace('_', '');
  });
};

export const sanitizeErrorFeedback = (input: {
  [key: string]: Array<string>;
}) => {
  // Loop at each error message and join them.
  // Also convert keys from snake_case to camelCase
  const newObject: any = {};

  Object.keys(input).forEach((key) => {
    newObject[toCamelCase(key)] = input[key].join();
  });

  return newObject;
};

export const capitalizeFirstLetter = (str: string) => {
  return str[0].toUpperCase() + str.slice(1);
};
