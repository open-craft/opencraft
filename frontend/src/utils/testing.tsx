import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
