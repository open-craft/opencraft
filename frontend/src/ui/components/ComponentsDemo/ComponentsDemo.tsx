import * as React from 'react';
import './styles.scss';
import { Table, Row, Col } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import { CollapseEditArea } from '../CollapseEditArea';
import { ColorInputField } from '../ColorInputField';
import { ContentPage } from '../ContentPage';
import { CustomStatusPill } from '../CustomStatusPill';
import { EmailActivationAlertMessage } from '../EmailActivationAlertMessage';
import { StepBar } from '../StepBar';
import { TextInputField } from '../TextInputField';
import messages from './displayMessages';

export const ComponentsDemo: React.FC = () => {
  const [inputColor, setInputColor] = React.useState('#000');
  const [inputTextValue, setInputTextValue] = React.useState('demo page');

  return (
    <div className="components-demo-page">
      <div className="components-demo-heading">
        <h1>Reusable Demo Components</h1>
      </div>
      <div className="components-demo-container">
        <div className="components-demo-text">
          The List of resuable components used inside the OCIM. Before
          generating a new component, check them.
        </div>
        <Table
          striped
          bordered
          hover
          className="components-demo-table-container"
        >
          <thead>
            <tr>
              <th>Component</th>
              <th>Description</th>
              <th>Example</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Collapse Edit Area</td>
              <td>
                Collapse Edit Area component will help create a editing area
                which can be expanded and collapsed based on the props. The Hero
                component has an use case of it.
              </td>
              <td className="collapse-edit-area">
                <CollapseEditArea>
                  <Row>
                    <Col key="exampleCollapseEditArea">
                      <WrappedMessage id="Demo Text" messages={messages} />
                    </Col>
                  </Row>
                </CollapseEditArea>
              </td>
            </tr>
            <tr>
              <td>Color Input Field</td>
              <td>
                Color Input Field component will help create a color input field
                where you can input your colors. The hero component as well as
                many other component has an use case of it.
              </td>
              <td className="color-input-field">
                <ColorInputField
                  fieldName="DemoExampleColor"
                  initialValue={inputColor}
                  messages={messages}
                  onChange={(fieldName: string, newColor: string) => {
                    setInputColor(newColor);
                  }}
                  hideTooltip
                />
              </td>
            </tr>
            <tr>
              <td>Content Page</td>
              <td>
                Content Page component will takes a title and some components as
                its children and provides you a div container with the component
                wrapped inside it. The login has an use case of it.
              </td>
              <td className="table-cell">
                <ContentPage title="Example Page">
                  <WrappedMessage id="Text" messages={messages} />
                </ContentPage>
              </td>
            </tr>
            <tr>
              <td>Custom Status Pill</td>
              <td>
                As the name suggests, the status pill component been used for
                showing instance changes with some colors and text. The
                Redeployment Toolbar has it which shows the progress of
                instance.
              </td>
              <td className="table-cell">
                <CustomStatusPill
                  loading
                  redeploymentStatus={null}
                  deploymentType={null}
                  cancelRedeployment={undefined}
                />
              </td>
            </tr>
            <tr>
              <td>Email Activation Alert Message</td>
              <td>
                This component wraps the WrappedMessage component for an email
                alert which you can see if you have not verified your registered
                email.
              </td>
              <td className="table-cell">
                <EmailActivationAlertMessage />
              </td>
            </tr>
            <tr>
              <td>Step Bar</td>
              <td>
                Step Bar component shows steps pending, registration pages has
                an use case for it.
              </td>
              <td className="table-cell">
                <StepBar count={3} currentStep={2} />
              </td>
            </tr>
            <tr>
              <td>Text Input Field</td>
              <td>A text input field component for inputing the texts.</td>
              <td className="table-cell">
                <TextInputField
                  fieldName="title"
                  messages={messages}
                  value={inputTextValue}
                  onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
                    setInputTextValue(event.target.value);
                  }}
                  reset={() => {
                    setInputTextValue('demo page');
                  }}
                  key="demo"
                />
              </td>
            </tr>
          </tbody>
        </Table>
      </div>
    </div>
  );
};
