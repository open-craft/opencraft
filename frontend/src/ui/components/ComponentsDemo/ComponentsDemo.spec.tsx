import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ComponentsDemo } from './ComponentsDemo';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<ComponentsDemo />).toJSON();
    expect(tree).toMatchSnapshot();
});
