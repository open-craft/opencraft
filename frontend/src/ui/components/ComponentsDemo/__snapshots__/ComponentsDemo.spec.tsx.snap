// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="components-demo-page"
>
  <div
    className="components-demo-heading"
  >
    <h1>
      Reusable Demo Components
    </h1>
  </div>
  <div
    className="components-demo-container"
  >
    <div
      className="components-demo-text"
    >
      The List of resuable components used inside the OCIM. Before generating a new component, check them.
    </div>
    <table
      className="components-demo-table-container table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Component
          </th>
          <th>
            Description
          </th>
          <th>
            Example
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Collapse Edit Area
          </td>
          <td>
            Collapse Edit Area component will help create a editing area which can be expanded and collapsed based on the props. The Hero component has an use case of it.
          </td>
          <td
            className="collapse-edit-area"
          >
            <div
              className="collapsible-edit-area"
            >
              <div
                className="edit-collapse"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  Edit
                  <i
                    className="fas xs fa-chevron-down"
                  />
                </a>
                <span
                  className="fill-line"
                />
              </div>
              <div
                aria-expanded={null}
                className="content collapse"
              >
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    Demo Text
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            Color Input Field
          </td>
          <td>
            Color Input Field component will help create a color input field where you can input your colors. The hero component as well as many other component has an use case of it.
          </td>
          <td
            className="color-input-field"
          >
            <div
              className="color-input form-group"
            >
              <label
                className="form-label"
              >
                DemoExampleColor
              </label>
              <div
                className="row"
              >
                <input
                  className="input-field-color form-control"
                  name="DemoExampleColor"
                  onClick={[Function]}
                  readOnly={true}
                  value="#000"
                />
                <div
                  className="input-field-preview"
                  style={
                    Object {
                      "backgroundColor": "#000",
                    }
                  }
                />
                <button
                  className="reset-value padded"
                  onClick={[Function]}
                  type="button"
                >
                  Reset
                </button>
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            Content Page
          </td>
          <td>
            Content Page component will takes a title and some components as its children and provides you a div container with the component wrapped inside it. The login has an use case of it.
          </td>
          <td
            className="table-cell"
          >
            <div
              className="content-page"
            >
              <div
                className="title-container"
              >
                <h1>
                  Example Page
                </h1>
              </div>
              <div
                className="content-page-container"
              >
                <div
                  className="content-page-content row"
                >
                  Text
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            Custom Status Pill
          </td>
          <td>
            As the name suggests, the status pill component been used for showing instance changes with some colors and text. The Redeployment Toolbar has it which shows the progress of instance.
          </td>
          <td
            className="table-cell"
          >
            <span
              className="status-pill badge badge-pill badge-primary"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <span
                className="dot"
                style={
                  Object {
                    "backgroundColor": "grey",
                  }
                }
              />
              <div
                className="text"
              >
                Status: loading
              </div>
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Email Activation Alert Message
          </td>
          <td>
            This component wraps the WrappedMessage component for an email alert which you can see if you have not verified your registered email.
          </td>
          <td
            className="table-cell"
          >
            <div
              className="d-flex justify-content-center align-middle email-activation-alert"
            >
              <i
                className="alert-icon fas fa-exclamation-triangle fa-xs"
              />
              One of your email addresses is pending confirmation. Please confirm it to begin managing your Open edX instance.
            </div>
          </td>
        </tr>
        <tr>
          <td>
            Step Bar
          </td>
          <td>
            Step Bar component shows steps pending, registration pages has an use case for it.
          </td>
          <td
            className="table-cell"
          >
            <div
              className="step-bar"
            >
              <svg
                className="step"
                viewBox="0 0 10 10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx={5}
                  cy={5}
                  fill="#6d9cae"
                  r={5}
                />
              </svg>
              <span
                className="line"
              />
              <svg
                className="circled-number"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  className="outer-circle"
                  cx={10}
                  cy={10}
                  r={10}
                />
                <circle
                  className="inner-circle"
                  cx={10}
                  cy={10}
                  r={7}
                  strokeWidth={2}
                />
                <text
                  className="circle-text"
                  color="#1e4c59"
                  dy="0.4em"
                  fontSize="10px"
                  strokeWidth="1px"
                  textAnchor="middle"
                  x="50%"
                  y="50%"
                >
                  2
                </text>
              </svg>
              <span
                className="line"
              />
              <svg
                className="step"
                viewBox="0 0 10 10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx={5}
                  cy={5}
                  fill="#6d9cae"
                  r={5}
                />
              </svg>
              <span
                className="line"
              />
            </div>
          </td>
        </tr>
        <tr>
          <td>
            Text Input Field
          </td>
          <td>
            A text input field component for inputing the texts.
          </td>
          <td
            className="table-cell"
          >
            <div
              className="text-input-container"
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  title
                </label>
                <input
                  className="form-control"
                  name="title"
                  onChange={[Function]}
                  value="demo page"
                />
              </div>
              <p>
                <button
                  className="reset-default"
                  onClick={[Function]}
                  type="button"
                >
                  Reset
                </button>
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
